### PR TITLE
[Codebase] const-like defines -> consts

### PR DESCRIPTION
**Pull request by Sh1ft0x0EF**

### CODE REVIEW

**Comment by WPMGPRoSToTeMa** [comment]
Pragma once isn't standardized yet.

**Comment by Sh1ft0x0EF** [comment]
And won't be I presume
But it works well and compilers can process headers a bit faster using it instead of guards

**Comment by Sh1ft0x0EF** [comment]
It's a bit strange that win compiler couldn't recognize constexpr
Some of your code is using features of C++11 but

**Comment by Sh1ft0x0EF** [comment]
You probably would be dead before pragma become standardized

File: rehlds/engine/hashpak.h [hunk 1]
```diff
@@ -32,7 +32,8 @@
 #include "custom.h"
 #include "FileSystem.h"
 
-constexpr auto HASHPAK_EXTENSION = ".hpk";
+constexpr char *HASHPAK_EXTENSION = ".hpk";
```
Comment (WPMGPRoSToTeMa) [inline rehlds/engine/hashpak.h]: Better change `char*` to `const char*` in needed function.

**Comment by LevShisterov** [comment]
Please split this in two: guard and other changes.

**Comment by Sh1ft0x0EF** [comment]
You wanna keep all the constexprs but restore guards? Or pragma with guards? Like that:

```cpp
#pragma once
#ifndef BLABLA_H
#define BLABLA_H
```

**Comment by WPMGPRoSToTeMa** [comment]
Pragma with guards should be good.

**Comment by Sh1ft0x0EF** [comment]
Alright

**Comment by LevShisterov** [comment]
Use two (or more) pull requests for these changes. It will make review easier and accept faster.

**Comment by Sh1ft0x0EF** [comment]
Win build has failed here (precompiled failed to be compiled because of constexpr usage)

**Comment by In-line** [comment]
What are advantages for `constexpr auto`?

**Comment by IgnacioFDM** [comment]
I'd say only using pragma once (without header guards) is better. It reduces the boilerplate, and the chance of making a mistake when writing the guards resulting in potential errors. Also you will never ever find a C++ compiler capable enough to support the C++11 stuff used here that doesn't support pragma once.

**Comment by Chuvi-w** [comment]
What's the main difference between _#define_ and _constexpr auto_?
Project is still compiliable in VS2013. _constexpr_ will break it.

**Comment by In-line** [comment]
@Chuvi-w I think compiler will generate debug info for `constexpr`, but will not for `#define`

**Comment by Chuvi-w** [comment]
@In-line do you really need debug info for thouse defines?

**Comment by WPMGPRoSToTeMa** [comment]
constexpr is type-safe.

**Comment by Chuvi-w** [comment]
@WPMGPRoSToTeMa Типа безопасный?

**Comment by Sh1ft0x0EF** [comment]
Finally!
Replaced const-like defines with consts for now (the only those which could be safely replaced)
I will replace them fully and with constexpr a bit later but this PR can be applied now

File: rehlds/engine/net.h [hunk 2]
```diff
@@ -91,25 +91,25 @@ enum
...
 
 // Flow control bytes per second limits
-#define MAX_RATE		100000.0f
-#define MIN_RATE		1000.0f
+const float MAX_RATE		= 100000.0f;
+const float MIN_RATE		= 1000.0f;
 
 // Default data rate
-#define DEFAULT_RATE	(9999.0f)
+const float DEFAULT_RATE	= (9999.0f);
```
Comment (In-line) [inline rehlds/engine/net.h]: There is no reason to use braces here

**Pull request merged**